internal/util/observability/observability.go

```diff
@@ -19,34 +19,82 @@ package observability
 
 import (
 	"context"
+	"errors"
 	"time"
 
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
 	otelsdkresource "go.opentelemetry.io/otel/sdk/resource"
 	otelsdktrace "go.opentelemetry.io/otel/sdk/trace"
 	otelsemconv "go.opentelemetry.io/otel/semconv/v1.21.0"
-
-	"github.com/FerretDB/FerretDB/internal/util/must"
+	"go.uber.org/zap"
 )
 
-// SetupOtel sets up OpenTelemetry exporter with fixed values.
-func SetupOtel(service string) func(context.Context) error {
-	exporter := must.NotFail(otlptracehttp.New(
+// OtelConfig is the configuration for OpenTelemetry.
+type OtelConfig struct {
+	Service  string
+	Version  string
+	Endpoint string
+}
+
+// Otel represents the OpenTelemetry system.
+type Otel struct {
```

Comment:
I'm not sure what an "OpenTelemetry system" is. I guess the official name is "[OTLP exporter](https://opentelemetry.io/docs/languages/go/exporters/#otlp)".

We could also do the same protection as in #4404, because we modify the global state with `otel.SetTracerProvider`